terraform/lambda/main.tf

```diff
@@ -0,0 +1,206 @@
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:BatchGetImage",
+          "ecr:BatchCheckLayerAvailability",
+          "ecr:GetAuthorizationToken"
+        ]
+        Resource = [
+          "arn:aws:ecr:${var.region}:${var.aws_account_id}:repository/${var.ecr_repository_name}",
+          "arn:aws:ecr:${var.region}:${var.aws_account_id}:repository/${var.ecr_repository_name}/*"
+        ]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "ecr:GetAuthorizationToken"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+  depends_on = [aws_iam_role.lambda_execution_role]
+}
+
+# 4. Add S3 access policy
+resource "aws_iam_role_policy" "lambda_s3_access" {
+  name = "${var.domain}-${var.service_subdomain}-lambda-s3-policy"
+  role = aws_iam_role.lambda_execution_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:PutObject",
+          "s3:GetObject",
+          "s3:ListBucket"
+        ]
+        Resource = [
+          "arn:aws:s3:::${var.tech_audit_data_bucket_name}",
+          "arn:aws:s3:::${var.tech_audit_data_bucket_name}/*"
+        ]
+      }
+    ]
+  })
+  depends_on = [aws_iam_role.lambda_execution_role]
+}
+
+# 5. Add additional permissions
+resource "aws_iam_role_policy" "lambda_additional_permissions" {
+  name = "${var.domain}-${var.service_subdomain}-policy-2"
+  role = aws_iam_role.lambda_execution_role.id
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "secretsmanager:GetSecretValue"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup"
+        ]
+        Resource = "arn:aws:logs:${var.region}:${var.aws_account_id}:*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = [
+          "arn:aws:logs:${var.region}:${var.aws_account_id}:log-group:/aws/lambda/${var.domain}-${var.service_subdomain}-lambda:*"
+        ]
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogGroup",
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "execute-api:Invoke",
+          "execute-api:ManageConnections"
+        ]
+        Resource = "arn:aws:execute-api:*:*:*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "s3:*",
+          "s3-object-lambda:*"
+        ]
+        Resource = "*"
+      }
+    ]
+  })
+  depends_on = [aws_iam_role.lambda_execution_role]
+}
+
+# 6. Create the Lambda function
+resource "aws_lambda_function" "tech_audit_lambda" {
```

Comment:
This needs to sit in the private VPC I believe.

@gibbardsteve  can you confirm?

If you take a look at the Policy and CoPilot Lambdas, the logic is there.

https://github.com/ONS-Innovation/github-copilot-usage-dashboard/blob/86f06704e1bc45ea45e4e4287eece4aa75d5c4d5/terraform/data_logger/main.tf#L25

Reply:
Yes - it would be better if we run everything that is not public facing in the private subnet of the VPC.

Reply:
Done [here](https://github.com/ONS-Innovation/keh-tech-audit-tool-api/pull/3/commits/60d69908e1a4c043007f404067677a14857ef7e4)